src/app/phase-bug-reporting/new-issue/new-issue.component.ts

```diff
@@ -53,6 +52,27 @@ export class NewIssueComponent implements OnInit {
         });
   }
 
+  canDeactivate() {
+    return !this.isTitleEditing() && !this.isDescriptionEditing()
+      && !this.isSeverityEditing() && !this.isTypeEditing();
+  }
+
+  isTitleEditing() {
+    return this.title.value !== '' && this.title.value !== null;
+  }
+
+  isDescriptionEditing() {
+    return this.description.value !== '' && this.description.value !== null;
+  }
+
+  isSeverityEditing() {
+    return this.severity.value !== '' && this.severity.value !== null;
+  }
+
+  isTypeEditing() {
+    return this.type.value !== '' && this.type.value !== null;
+  }
+
```

Comment:
I feel that the logic here is quite repetitive, just on different attributes. Maybe we can apply the same standardised check on these attributes?